src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -230,7 +230,6 @@ private ExtensionInfo MakeDeclaredExtensionInfo(ConsList<TypeSymbol>? basesBeing
             TypeSymbol? underlyingType = null;
             SourceLocation? underlyingTypeLocation = null;
 
-            Debug.Assert(basesBeingResolved == null || !basesBeingResolved.ContainsReference(this.OriginalDefinition));
```

Comment:
>Debug.Assert(basesBeingResolved == null || !basesBeingResolved.ContainsReference(this.OriginalDefinition)); [](http://example.com/codeflow?start=12&length=107)

Why is this assert no longer valid and why it is Ok to continue if the assert fails?